src/v/archival/segment_reupload.cc

```diff
@@ -498,6 +498,35 @@ segment_collector::make_upload_candidate(
     auto head_seek = head_seek_result.value();
     auto tail_seek = tail_seek_result.value();
 
+    if (head_seek.offset_inside_batch || tail_seek.offset_inside_batch) {
+        vlog(
+          archival_log.warn,
```

Comment:
Fine with this as is, but perhaps we should make this an error later so we can catch more things and understand the scenario better